src/promptflow/promptflow/contracts/tool.py

```diff
@@ -185,14 +188,28 @@ def is_connection_value(val: Any) -> bool:
         return val in connections.values() or ConnectionType.is_custom_strong_type(val)
 
     @staticmethod
-    def is_custom_strong_type(val):
-        """Check if the given value is a custom strong type connection."""
+    def is_custom_strong_type(val: Any) -> bool:
+        """Check if the given value is a custom strong type connection.
+
+        :param val: The value to check
+        :type val: Any
+        :return: Whether the given value is a custom strong type
+        :rtype: bool
+        """
+
         from promptflow._sdk.entities import CustomStrongTypeConnection
 
-        # TODO: replace the hotfix "try-except" with a more graceful solution."
+        val = type(val) if not isinstance(val, type) else val
+        # Check for instances of GenericAlias (for parameterized generic types like list[str])
+        if isinstance(val, types.GenericAlias):
+            return False
+
         try:
             return issubclass(val, CustomStrongTypeConnection)
-        except Exception:
+        except TypeError as e:
+            # TypeError is not expected to happen, but if it does, we will log it for debugging and return False.
+            # The try-except block cannot be confidently removed due to the uncertainty of TypeError that may occur.
```

Comment:
Why not capture other type exception and log?

Reply:
According to this doc, it seems the only exception could be TypeError:
![image](https://github.com/microsoft/promptflow/assets/46446115/42954c0f-e63c-47c8-9202-f8aec931def0)
https://docs.python.org/3/library/functions.html

And in our promptflow code, nowhere overrides " __subclasscheck__()", so I think there coud be no other exception type